examples/custom-service.mdx

```diff
@@ -20,71 +38,163 @@ edition = "2021"
 [lib]
 
 [dependencies]
-shuttle-service = { version = "0.9.0" }
-shuttle-shared-db = { version = "0.9.0", features = ["postgres"] }
-sqlx = { version = "0.6", features = ["runtime-tokio-native-tls", "postgres"] }
+anyhow = "1.0.62"
+axum = "0.6.4"
+hyper = "0.14.24"
+poise = "0.5.2"
+serde = "1.0"
+shuttle-secrets = "0.9.0"
+shuttle-service = { version = "0.9.0"}
+sync_wrapper = "0.1.2"
+tokio = "1.22.0"
 ```
 
-This section will explain how to create a custom shuttle service. To do this we need 
-to return a struct from our `shuttle_service::main` function that implements `shuttle_service::Service`.
-
-In this simple example we will implement `Service` for a `PgPool`, that will simply return 'Hello world'
-from the database on startup. In Rust we can't implement foreign traits for foreign types, so we have 
-to create a wrapper around our `PgPool`.
+### Getting Started
+To get started, we need to return a wrapper struct from our `shuttle_service::main` function that 
+implements `shuttle_service::Service`, which we can declare like so:
 
-```rust
-struct PoolService {
-    pool: sqlx::PgPool,
+```rust lib.rs
+pub struct CustomService {
+    discord_bot:
+        FrameworkBuilder<Data, Box<(dyn std::error::Error +
+         std::marker::Send + Sync + 'static)>>,
+    router: SyncWrapper<Router>,
 }
 ```
 
 Then we need to implement `shuttle_service::Service` for our wrapper. If you need to bind to an address,
 for example if you're implementing service for an HTTP server, you can use the `addr` argument from `bind`.
 You can only have one HTTP service bound to the `addr`, but you can start other services that don't rely on 
-binding to a socket. A common usecase is starting an HTTP server like `axum`, alongside a `serenity` bot 
-that doesn't need to bind to a socket. Then you would include both in the same wrapper struct, and start 
-them as normal in the `bind` function.
+binding to a socket.
+
+We can do do this by simply adding an implementation for our wrapper struct containing a function called `start`,
+which will contain all of our tasks that we want to run and simply run them all together in one `tokio::select!` macro
+(currently, this code doesn't work due to no base implementation - we will implement this later on):
 
 ```rust lib.rs
 #[shuttle_service::async_trait]
-impl shuttle_service::Service for PoolService {
+impl shuttle_service::Service for CustomService {
     async fn bind(
         mut self: Box<Self>,
-        _addr: std::net::SocketAddr,
+        addr: std::net::SocketAddr,
     ) -> Result<(), shuttle_service::error::Error> {
-        self.start().await?;
+        self.start(addr).await.expect("Something went wrong!");
```

Comment:
While this is nice, I'm wondering if we shouldn't implement a `start` function for the custom service, and rather do it directly here. This way people who just want to know how to create a custom service can see the core of how to do it early, and then read how to create the router and the bot if they want.

Reply:
Hmmm, yeah probably. That makes more sense in terms of the actual writing as when I was trying to structure the writing it was a bit awkward.